### ISSUE TITLE
Mouse cursor visible in swaylock

### ISSUE THREAD

**Issue by fkranhold** [swaywm/swaylock#333]
This might be related to #273, but it seems to appear in a newer version and behaves slightly differently (see below):

I am running sway 1.8.1 with swaylock 1.7.2 on Arch. When locking the screen via `swaylock`, the mouse cursor remains visible. Once I move the cursor slightly, it vanishes. The same issue does not occur with sway 1.7-6 and swaylock 1.7.2 on Debian stable.

**Comment by bitterhalt** [comment]
Same behavior here. 
Sway : 1.8.1-4
Swaylock: 1.7.2-1

**Comment by cnschn** [comment]
Exactly the same behavior for me.

Arch, `swaylock 1.7.2-1`, `swayfx 0.3.2-5` (based on `sway 1.8.1`)

**Comment by kennylevinsen** [comment]
This might be a sway issue.

Please do not post "me too" comments, especially for unsupported forks.

**Comment by cnschn** [comment]
Fair point regarding swayfx - I just tested it on Hyprland, swaylock shows the same behavior there.

**Comment by M3chanism** [comment]
Exact same behavior on NixOS using Hyprland. Mouse cursor remains visible then vanishes with slight movement.
Hyprland : 34.0
Swaylock: 1.7.2
Distro: Nixos

**Comment by kennylevinsen** [comment]
`swaylock` hides the cursor when it gets pointer focus, it does not do anything on movement.

The likely compositor bug would be that focus is not being recomputed correctly when the lock surface opens, instead waiting for the usual mouse move check.

**Comment by emersion** [comment]
Yeah, I don't think there's anything to be done in swaylock here. Please report a bug to your compositor if you still see this happening.
